File: Python/day1.py
import sys
import math

def depth_increases(depths: list[int]) -> int:
    increases = 0
    for i in range (0, len(depths) - 1):
        if depths[i] < depths[i+1]:
            increases += 1
    return increases

def depth_increases_sliding_windows(depths: list[int]) -> int:
    increases = 0
    prev_window = math.inf
    for i in range(1, len(depths) - 1):
        window = (depths[i-1] + depths[i] + depths[i+1])
        if (window > prev_window):
            increases += 1
        prev_window = window 
    return increases

if __name__ == "__main__":
    # python day1.py < day1_puzzle.txt into cmd promt
    lines = sys.stdin.readlines()
    depths = [int(line) for line in lines]
    solution = depth_increases(depths)
    print("First solution:", solution)

    solution_two = depth_increases_sliding_windows(depths)
    print ("Second solution:", solution_two)

File: C++/day1.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

int depthIncreases (const vector<int> &depths) {
    const int depthsSize = depths.size();
    int increases = 0;
    for (int i = 0; i < depthsSize - 1; i++) {
        if (depths[i] < depths[i+1]) {
            increases++;
        }
    }
    return increases;
}

int depthIncreasesSlidingWindow(const vector<int> &depths) {
    const int depthsSize = depths.size();
    int increases = 0;
    int prevWindow = (depths[0] + depths[1] + depths[2]);
    for (int i = 2; i < depthsSize - 1; i++) {
        int window = (depths[i-1] + depths[i] + depths[i+1]);
        if (window > prevWindow) {
            increases++;
        }
        prevWindow = window;
    }
    return increases;
}


int main() {

    // get lines from txt file
    vector<int> depths;
    for (string line; getline(std::cin, line);) {
        depths.push_back(stoi(line));
    }
    
    int solution = depthIncreases(depths);
    cout << "First solution: " << solution << endl;

    int solutionTwo = depthIncreasesSlidingWindow(depths);
    cout << "Second solution: "<< solutionTwo << endl;

    return 0;
}
